lib/index.js

```diff
@@ -0,0 +1,76 @@
+var libvirt_native = require('bindings')('libvirt');
+
+/**
+ * This function should be called first to get a connection to the
+ * Hypervisor and xen store
+ * 
+ * If @name is None, if the LIBVIRT_DEFAULT_URI environment variable is set,
+ * then it will be used. Otherwise if the client configuration file
+ * has the "uri_default" parameter set, then it will be used. Finally
+ * probing will be done to determine a suitable default driver to activate.
+ * This involves trying each hypervisor in turn until one successfully opens.
+ * 
+ * If connecting to an unprivileged hypervisor driver which requires
+ * the libvirtd daemon to be active, it will automatically be launched
+ * if not already running. This can be prevented by setting the
+ * environment variable LIBVIRT_AUTOSTART=0
+ * 
+ * URIs are documented at http://libvirt.org/uri.html
+ * 
+ * virConnectClose should be used to release the resources after the connection
+ * is no longer needed.
+ *
+ * @param {String} name
+ * @return {Object} virConnect
+ */
+function open(name) {
+    let ret = libvirt_native.virConnectOpen(name);
+    if(!ret) throw new Error('virConnectOpen() failed');
+    return new virConnect(ret);
+}
+
+class virConnect {
+    constructor(conn) {
+        this.conn = conn;
+    }
+
+    /**
+     * Try to lookup a domain on the given hypervisor based on its name.
+     *
+     * virDomainFree should be used to free the resources after the
+     * domain object is no longer needed.
+     *
+     * @param {String} name
+     * @returns {Object} virDomain
+     */
+    lookupByName(name) {
+        let ret = libvirt_native.virDomainLookupByName(this.conn, name);
+        if(!ret) throw new Error('virDomainLookupByName() failed');
+        return new virDomain(this, ret);
+    }
+}
+
+class virDomain {
+    constructor(conn, domain) {
+        this.conn = conn;
+        this.domain = domain;
+    }
+
+    /**
+     * Get the hypervisor ID number for the domain
+     *
+     * @return {Integer}
+     */
+    getId() {
```

Comment:
Is there a good reason for that? I feel virDomainGetId -> getId (and eg. virDomainOpenChannel -> openChannel) is more consistent

Reply:
Reason? Yes. Good one? I don't know. It depends on the perception of app developers. In (python bindings)[https://libvirt.org/python.html] we do that. It feels more idiomatic to have ID as the name, instead of getID, which sounds like a getter and there is no need for getters in python. Unfortunately the bindings are old and we can't change much, so it still is a function and not a member with an assigned getter and setter, setters are just setBlahBlah() functions (the prefix is not removed). On the other hand for our getters you sometimes need to pass some flags, so we might not use that approach in JS either. I'm not sure how to approach this. I initially thought at least naming things the same as in python would be nice, but... I don't know. Have a look at the python description and try to come to a conclusion backed up with some reasoning. I mean for now we have either a) consisitent with C APIs or b) consistent with Python APIs.